### PR TITLE
Add engine request builder factory

### PR DESCRIPTION
**Pull request by sergeymirzoyan1991**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -38,7 +38,9 @@ cloudChiprCli
 try {
   command.parseAsync(process.argv).catch(e => {
     console.error(chalk.red(chalk.underline('Error:'), e.message))
+    console.error(e.stack)
```
Comment (marashlyansuren) [inline src/index.ts:41]: @sergeymirzoyan1991 jan maybe we need to hide this behind some option `--vvv`

Comment (sergeymirzoyan1991) [reply src/index.ts:41]: my bad, i should have removed this
removing now

Comment (sergeymirzoyan1991) [reply src/index.ts]: removed

File: src/requests/filter-provider.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+export class FilterProvider {
+  static getCollectFilter (options: OptionValues, subCommand: SubCommandInterface): FilterInterface {
+    const builder = new FilterBuilder(new FilterValidator(subCommand))
+    return builder
+      .load(yaml.parse(readFileSync(options.filter, 'utf8')))
+      .toList()
+  }
+
+  static getCleanFilter (ids: string[], subCommand: SubCommandInterface): FilterInterface {
```
Comment (marashlyansuren) [inline src/requests/filter-provider.ts:15]: So is this already in use???

Comment (sergeymirzoyan1991) [reply src/requests/filter-provider.ts:15]: this is the skeleton of the clean command, the next PR will start using this

File: src/requests/engine-collect-request-builder.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import { OptionValues } from 'commander'
+import { FilterProvider } from './filter-provider'
+import { Verbose } from '../constants'
+import EngineRequestBuilder from './engine-request-builder'
+
+export default class EngineCollectRequestBuilder extends EngineRequestBuilder {
+  private options: OptionValues;
+
+  // Only enabled Regions
+  private allRegions: Set<string> = new Set([
```
Comment (marashlyansuren) [inline src/requests/engine-collect-request-builder.ts:15]: maybe we can get them from some config/constat file

**Review by marashlyansuren — APPROVED** [review]
left small comment, but good to go

**Pull request merged**